apps/ejabberd/src/rdbms/mongoose_rdbms.erl

```diff
@@ -473,7 +473,7 @@ connect(Settings, Retry, RetryAfter, MaxRetryDelay) ->
                        " Retrying in ~p seconds.", [Settings, Error, SleepFor]),
             timer:sleep(timer:seconds(SleepFor)),
             NextRetryDelay = RetryAfter * RetryAfter,
-            connect(Settings, Retry - 1, max(MaxRetryDelay, NextRetryDelay), MaxRetryDelay)
+            connect(Settings, Retry - 1, min(MaxRetryDelay, NextRetryDelay), MaxRetryDelay)
```

Comment:
This line is not covered by tests at all. I'm pretty sure the bug would not appear if there was test for that. A regression test for this is needed.